### ISSUE TITLE
hevc_vaapi in nzbtomedia error

### ISSUE THREAD

**Issue by rosskopf** [clinton-hall/nzbToMedia#1323]
Hi,

I love this script and it works well, but then i wanted to use hevc_vaapi for faster encoding.
when i run ffmeg manual 
`/home/benjamin/bin/ffmpeg -hwaccel vaapi -hwaccel_output_format vaapi -fflags +genpts -i /media/externeHD/transmission/process/tv/The.Big.Bang.Theory.S11E12.iNTERNAL.720p.WEB.x264-BAMBOOZLE[eztv].mkv/The.Big.Bang.Theory.S11E12.iNTERNAL.720p.WEB.x264-BAMBOOZLE[eztv].mkv -map 0:0 -map 0:1 -c:v hevc_vaapi -preset slow -crf 19 -c:a:0 copy -sn /media/externeHD/transmission/series/test.mkv`

> ffmpeg version 3.4.1 Copyright (c) 2000-2017 the FFmpeg developers
  built with gcc 5.4.0 (Ubuntu 5.4.0-6ubuntu1~16.04.5) 20160609
  configuration: --prefix=/home/benjamin/ffmpeg_build --pkg-config-flags=--static --extra-cflags=-I/home/benjamin/ffmpeg_build/include --extra-ldflags=-L/home/benjamin/ffmpeg_build/lib --extra-libs='-lpthread -lm' --bindir=/home/benjamin/bin --enable-gpl --enable-libass --enable-libfdk-aac --enable-libfreetype --enable-libmp3lame --enable-libopus --enable-libtheora --enable-libvorbis --enable-libx264 --enable-libx265 --enable-vaapi --enable-nonfree
  libavutil      55. 78.100 / 55. 78.100
  libavcodec     57.107.100 / 57.107.100
  libavformat    57. 83.100 / 57. 83.100
  libavdevice    57. 10.100 / 57. 10.100
  libavfilter     6.107.100 /  6.107.100
  libswscale      4.  8.100 /  4.  8.100
  libswresample   2.  9.100 /  2.  9.100
  libpostproc    54.  7.100 / 54.  7.100
Input #0, matroska,webm, from '/media/externeHD/transmission/process/tv/The.Big.Bang.Theory.S11E12.iNTERNAL.720p.WEB.x264-BAMBOOZLE[eztv].mkv/The.Big.Bang.Theory.S11E12.iNTERNAL.720p.WEB.x264-BAMBOOZLE[eztv].mkv':
  Metadata:
    encoder         : libebml v1.3.3 + libmatroska v1.4.4
    creation_time   : 2018-01-05T04:35:57.000000Z
  Duration: 00:20:27.61, start: 0.000000, bitrate: 3007 kb/s
    Stream #0:0: Video: h264 (High), yuv420p(progressive), 1280x720 [SAR 1:1 DAR 16:9], 23.98 fps, 23.98 tbr, 1k tbn, 47.95 tbc (default)
    Metadata:
      BPS             : 2880190
      BPS-eng         : 2880190
      DURATION        : 00:20:27.560000000
      DURATION-eng    : 00:20:27.560000000
      NUMBER_OF_FRAMES: 29432
      NUMBER_OF_FRAMES-eng: 29432
      NUMBER_OF_BYTES : 441950804
      NUMBER_OF_BYTES-eng: 441950804
      _STATISTICS_WRITING_APP: mkvmerge v8.8.0 ('Wind at my back') 64bit
      _STATISTICS_WRITING_APP-eng: mkvmerge v8.8.0 ('Wind at my back') 64bit
      _STATISTICS_WRITING_DATE_UTC: 2018-01-05 04:35:57
      _STATISTICS_WRITING_DATE_UTC-eng: 2018-01-05 04:35:57
      _STATISTICS_TAGS: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES
      _STATISTICS_TAGS-eng: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES
    Stream #0:1: Audio: aac (LC), 48000 Hz, stereo, fltp (default)
    Metadata:
      BPS             : 125375
      BPS-eng         : 125375
      DURATION        : 00:20:27.605000000
      DURATION-eng    : 00:20:27.605000000
      NUMBER_OF_FRAMES: 57544
      NUMBER_OF_FRAMES-eng: 57544
      NUMBER_OF_BYTES : 19238878
      NUMBER_OF_BYTES-eng: 19238878
      _STATISTICS_WRITING_APP: mkvmerge v8.8.0 ('Wind at my back') 64bit
      _STATISTICS_WRITING_APP-eng: mkvmerge v8.8.0 ('Wind at my back') 64bit
      _STATISTICS_WRITING_DATE_UTC: 2018-01-05 04:35:57
      _STATISTICS_WRITING_DATE_UTC-eng: 2018-01-05 04:35:57
      _STATISTICS_TAGS: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES
      _STATISTICS_TAGS-eng: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES
File '/media/externeHD/transmission/series/test.mkv' already exists. Overwrite ? [y/N] y
libva info: VA-API version 0.39.0
libva info: va_getDriverName() returns 0
libva info: Trying to open /usr/lib/x86_64-linux-gnu/dri/i965_drv_video.so
libva info: Found init function __vaDriverInit_0_39
libva info: va_openDriver() returns 0
Stream mapping:
  Stream #0:0 -> #0:0 (h264 (native) -> hevc (hevc_vaapi))
  Stream #0:1 -> #0:1 (copy)
Press [q] to stop, [?] for help
Output #0, matroska, to '/media/externeHD/transmission/series/test.mkv':
  Metadata:
    encoder         : Lavf57.83.100
    Stream #0:0: Video: hevc (hevc_vaapi) (Main), vaapi_vld, 1280x720 [SAR 1:1 DAR 16:9], q=2-31, 23.98 fps, 1k tbn, 23.98 tbc (default)
    Metadata:
      BPS             : 2880190
      BPS-eng         : 2880190
      DURATION        : 00:20:27.560000000
      DURATION-eng    : 00:20:27.560000000
      NUMBER_OF_FRAMES: 29432
      NUMBER_OF_FRAMES-eng: 29432
      NUMBER_OF_BYTES : 441950804
      NUMBER_OF_BYTES-eng: 441950804
      _STATISTICS_WRITING_APP: mkvmerge v8.8.0 ('Wind at my back') 64bit
      _STATISTICS_WRITING_APP-eng: mkvmerge v8.8.0 ('Wind at my back') 64bit
      _STATISTICS_WRITING_DATE_UTC: 2018-01-05 04:35:57
      _STATISTICS_WRITING_DATE_UTC-eng: 2018-01-05 04:35:57
      _STATISTICS_TAGS: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES
      _STATISTICS_TAGS-eng: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES
      encoder         : Lavc57.107.100 hevc_vaapi
    Stream #0:1: Audio: aac (LC) ([255][0][0][0] / 0x00FF), 48000 Hz, stereo, fltp (default)
    Metadata:
      BPS             : 125375
      BPS-eng         : 125375
      DURATION        : 00:20:27.605000000
      DURATION-eng    : 00:20:27.605000000
      NUMBER_OF_FRAMES: 57544
      NUMBER_OF_FRAMES-eng: 57544
      NUMBER_OF_BYTES : 19238878
      NUMBER_OF_BYTES-eng: 19238878
      _STATISTICS_WRITING_APP: mkvmerge v8.8.0 ('Wind at my back') 64bit
      _STATISTICS_WRITING_APP-eng: mkvmerge v8.8.0 ('Wind at my back') 64bit
      _STATISTICS_WRITING_DATE_UTC: 2018-01-05 04:35:57
      _STATISTICS_WRITING_DATE_UTC-eng: 2018-01-05 04:35:57
      _STATISTICS_TAGS: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES
      _STATISTICS_TAGS-eng: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES
frame= 9436 fps=121 q=-0.0 Lsize=  118133kB time=00:06:33.91 bitrate=2456.7kbits/s speed=5.07x

It works, so i thought i edit the script so that the ffmpeg command is the same

> [Transcoder]
    # getsubs. enable to download subtitles.
    getSubs = 1
    # subLanguages. create a list of languages in the order you want them in your subtitles.
    subLanguages = "nl,en"
    # transcode. enable to use transcoder
    transcode = 1
    ###### duplicate =1 will create a new file. =0 will replace the original
    duplicate = 0
    # concat. joins cd1 cd2 etc into a single video.
    concat = 1
    # IgnoreExtensions is a comma-separated list of extensions that will not be transcoded.
    ignoreExtensions = ""
    # outputFastStart. 1 will use -movflags + faststart. 0 will disable this from being used.
    outputFastStart = 0
    # outputQualityPercent. used as -q:a value. 0 will disable this from being used.
    outputQualityPercent = 0
    # outputVideoPath. Set path you want transcoded videos moved to. Leave blank to disable.
    outputVideoPath = /media/externeHD/transmission/series
    # processOutput. 1 will send the outputVideoPath to SickBeard/CouchPotato. 0 will send original files.
    processOutput = 1
    # audioLanguage. set the 3 letter language code you want as your primary audio track.
    audioLanguage = eng
    # allAudioLanguages. 1 will keep all audio tracks (uses AudioCodec3) where available.
    allAudioLanguages = 0
    # allSubLanguages. 1 will keep all existing sub languages. 0 will discard those not in your list above.
    allSubLanguages = 0
    # embedSubs. 1 will embed external sub/srt subs into your video if this is supported.
    embedSubs = 0
    # burnInSubtitle. burns the default sub language into your video (needed for players that don't support subs)
    burnInSubtitle = 0
    # extractSubs. 1 will extract subs from the video file and save these as external srt files.
    extractSubs = 0
    # externalSubDir. set the directory where subs should be saved (if not the same directory as the video)
    externalSubDir = ""
    # hwAccel. 1 will set ffmpeg to enable hardware acceleration (this requires a recent ffmpeg)
    hwAccel = 0
    # generalOptions. Enter your additional ffmpeg options here with commas to separate each option/value (i.e replace spaces with commas).
    generalOptions = "-hwaccel vaapi,-hwaccel_output_format vaapi"
    # outputDefault. Loads default configs for the selected device. The remaining options below are ignored.
    # If you want to use your own profile, leave this blank and set the remaining options below.
    # outputDefault profiles allowed: iPad, iPad-1080p, iPad-720p, Apple-TV2, iPod, iPhone, PS3, xbox, Roku-1080p, Roku-720p, Roku-480p, mkv, mp4-scene-release
    outputDefault = ""
    #### Define custom settings below.
    outputVideoExtension = .mkv
    outputVideoCodec = hevc_vaapi
    VideoCodecAllow = "HEVC,hevc,h265,x265,h.265"
    outputVideoPreset = ultrafast
    outputVideoResolution = ""
    outputVideoFramerate = ""
    outputVideoBitrate = ""
    outputVideoCRF = 19
    outputVideoLevel = ""
    outputAudioCodec = ac3
    AudioCodecAllow = "DTS,ac3,libfdk_aac,a_aac,aac,libfaac,a_ac3,ac-3"

i added this to generalOptions = "-hwaccel vaapi,-hwaccel_output_format vaapi"
because hwaccel auto giving me errors in the manual run like above.

Now, when i download a episode and Torrenttomedia.py gets called i get this error

> 2018-01-06 13:40:39 INFO    ::MAIN: Loading config from [/home/benjamin/nzbToMedia/autoProcessMedia.cfg]
2018-01-06 13:40:39 INFO    ::ENVIRONMENT: LESSOPEN: | /usr/bin/lesspipe %s
2018-01-06 13:40:39 INFO    ::ENVIRONMENT: SSH_CLIENT: 192.168.0.16 56042 2220
2018-01-06 13:40:39 INFO    ::ENVIRONMENT: LOGNAME: benjamin
2018-01-06 13:40:39 INFO    ::ENVIRONMENT: USER: benjamin
2018-01-06 13:40:39 INFO    ::ENVIRONMENT: PATH: /home/benjamin/bin:/home/benjamin/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/snap/bin
2018-01-06 13:40:39 INFO    ::ENVIRONMENT: HOME: /home/benjamin
2018-01-06 13:40:39 INFO    ::ENVIRONMENT: DISPLAY: server/unix:0
2018-01-06 13:40:39 INFO    ::ENVIRONMENT: LANG: nl_BE.UTF-8
2018-01-06 13:40:39 INFO    ::ENVIRONMENT: TERM: xterm-256color
2018-01-06 13:40:39 INFO    ::ENVIRONMENT: SHELL: /bin/bash
2018-01-06 13:40:39 INFO    ::ENVIRONMENT: LANGUAGE: nl_BE:nl
2018-01-06 13:40:39 INFO    ::ENVIRONMENT: SHLVL: 1
2018-01-06 13:40:39 INFO    ::ENVIRONMENT: XDG_RUNTIME_DIR: /run/user/1000
2018-01-06 13:40:39 INFO    ::ENVIRONMENT: SSH_AUTH_SOCK: /tmp/ssh-wAz6Y0uArL/agent.19803
2018-01-06 13:40:39 INFO    ::ENVIRONMENT: XDG_SESSION_ID: 1068
2018-01-06 13:40:39 INFO    ::ENVIRONMENT: _: ./TorrentToMedia.py
2018-01-06 13:40:39 INFO    ::ENVIRONMENT: SSH_CONNECTION: 192.168.0.16 56042 192.168.0.21 2220
2018-01-06 13:40:39 INFO    ::ENVIRONMENT: LESSCLOSE: /usr/bin/lesspipe %s %s
2018-01-06 13:40:39 INFO    ::ENVIRONMENT: SSH_TTY: /dev/pts/6
2018-01-06 13:40:39 INFO    ::ENVIRONMENT: OLDPWD: /home/benjamin
2018-01-06 13:40:39 INFO    ::ENVIRONMENT: XDG_DATA_DIRS: /usr/local/share:/usr/share:/var/lib/snapd/desktop
2018-01-06 13:40:39 INFO    ::ENVIRONMENT: PWD: /home/benjamin/nzbToMedia
2018-01-06 13:40:39 INFO    ::ENVIRONMENT: MAIL: /var/mail/benjamin
2018-01-06 13:40:39 INFO    ::ENVIRONMENT: LS_COLORS: rs=0:di=01;34:ln=01;36:mh=00:pi=40;33:so=01;35:do=01;35:bd=40;33;01:cd=40;33;01:or=40;31;01:mi=00:su=37;41:sg=30;43:ca=30;41:tw=30;42:ow=34;42:st=37;44:ex=01;32:*.tar=01;31:*.tgz=01;31:*.arc=01;31:*.arj=01;31:*.taz=01;31:*.lha=01;31:*.lz4=01;31:*.lzh=01;31:*.lzma=01;31:*.tlz=01;31:*.txz=01;31:*.tzo=01;31:*.t7z=01;31:*.zip=01;31:*.z=01;31:*.Z=01;31:*.dz=01;31:*.gz=01;3$
2018-01-06 13:40:39 INFO    ::MAIN: Checking database structure...
2018-01-06 13:40:39 DEBUG   ::MAIN: Checking Initial Schema database upgrade
2018-01-06 13:40:39 DB      ::MAIN: nzbtomedia.db: SELECT 1 FROM sqlite_master WHERE name = ?; with args ('db_version',)
2018-01-06 13:40:39 DB      ::MAIN: nzbtomedia.db: SELECT db_version FROM db_version
2018-01-06 13:40:39 DEBUG   ::MAIN: InitialSchema upgrade not required
2018-01-06 13:40:39 DEBUG   ::MAIN: Checking if we can use git commands: git version
2018-01-06 13:40:39 DEBUG   ::MAIN: Executing git version with your shell in /home/benjamin/nzbToMedia
2018-01-06 13:40:39 DEBUG   ::MAIN: git output: git version 2.7.4
2018-01-06 13:40:39 DEBUG   ::MAIN: git version : returned successful
2018-01-06 13:40:39 DEBUG   ::MAIN: Using: git
2018-01-06 13:40:39 DEBUG   ::MAIN: Executing git symbolic-ref -q HEAD with your shell in /home/benjamin/nzbToMedia
2018-01-06 13:40:39 DEBUG   ::MAIN: git output: refs/heads/master
2018-01-06 13:40:39 DEBUG   ::MAIN: git symbolic-ref -q HEAD : returned successful
2018-01-06 13:40:39 INFO    ::MAIN: Checking if git needs an update
2018-01-06 13:40:39 DEBUG   ::MAIN: Executing git rev-parse HEAD with your shell in /home/benjamin/nzbToMedia
2018-01-06 13:40:39 DEBUG   ::MAIN: git output: ee243fbba51e66ed12ad17d57e016b4db5bdc720
2018-01-06 13:40:39 DEBUG   ::MAIN: git rev-parse HEAD : returned successful
2018-01-06 13:40:39 DEBUG   ::MAIN: Executing git fetch origin with your shell in /home/benjamin/nzbToMedia
2018-01-06 13:40:42 DEBUG   ::MAIN: git output:
2018-01-06 13:40:42 DEBUG   ::MAIN: git fetch origin : returned successful
2018-01-06 13:40:42 DEBUG   ::MAIN: Executing git rev-parse --verify --quiet "@{upstream}" with your shell in /home/benjamin/nzbToMedia
2018-01-06 13:40:42 DEBUG   ::MAIN: git output: ee243fbba51e66ed12ad17d57e016b4db5bdc720
2018-01-06 13:40:42 DEBUG   ::MAIN: git rev-parse --verify --quiet "@{upstream}" : returned successful
2018-01-06 13:40:42 DEBUG   ::MAIN: Executing git rev-list --left-right "@{upstream}"...HEAD with your shell in /home/benjamin/nzbToMedia
2018-01-06 13:40:42 DEBUG   ::MAIN: git output:
2018-01-06 13:40:42 DEBUG   ::MAIN: git rev-list --left-right "@{upstream}"...HEAD : returned successful
2018-01-06 13:40:42 DEBUG   ::MAIN: cur_commit = ee243fbba51e66ed12ad17d57e016b4db5bdc720 % (newest_commit)= ee243fbba51e66ed12ad17d57e016b4db5bdc720, num_commits_behind = 0, num_commits_ahead = 0
2018-01-06 13:40:42 INFO    ::MAIN: No update needed
2018-01-06 13:40:42 INFO    ::MAIN: nzbToMedia Version:ee243fbba51e66ed12ad17d57e016b4db5bdc720 Branch:master (Linux 4.4.0-104-generic)
2018-01-06 13:40:42 WARNING ::MAIN: Failed to locate par2. Repair and rename using par files will not be possible!
2018-01-06 13:40:42 DEBUG   ::MAIN: Connecting to transmission: http://localhost:9099
2018-01-06 13:40:42 INFO    ::MAIN: #########################################################
2018-01-06 13:40:42 INFO    ::MAIN: ## ..::[TorrentToMedia.py]::.. ##
2018-01-06 13:40:42 INFO    ::MAIN: #########################################################
2018-01-06 13:40:42 DEBUG   ::MAIN: Options passed into TorrentToMedia: ['./TorrentToMedia.py']
2018-01-06 13:40:42 WARNING ::MAIN: Invalid number of arguments received from client, Switching to manual run mode ...
2018-01-06 13:40:42 INFO    ::MAIN: Searching /media/externeHD/transmission/series for mediafiles to post-process ...
2018-01-06 13:40:42 DEBUG   ::MAIN: Found file test.mkv in root directory /media/externeHD/transmission/series.
2018-01-06 13:40:42 INFO    ::COPYLINK: MEDIAFILE: [test.mkv]
2018-01-06 13:40:42 INFO    ::COPYLINK: SOURCE FOLDER: [/media/externeHD/transmission/series]
2018-01-06 13:40:42 INFO    ::COPYLINK: TARGET FOLDER: [/media/externeHD/transmission/series/transmission]
2018-01-06 13:40:42 INFO    ::COPYLINK: Hard linking SOURCE MEDIAFILE -> TARGET FOLDER
2018-01-06 13:40:42 INFO    ::MAIN: Starting manual run for NzbDrone:tv - Folder:/media/externeHD/transmission/series/transmission
2018-01-06 13:40:42 INFO    ::MAIN: Checking database for download info for transmission ...
2018-01-06 13:40:42 DB      ::DB: Getting download info for transmission from the DB
2018-01-06 13:40:42 DB      ::MAIN: nzbtomedia.db: SELECT * FROM downloads WHERE input_name=? AND status=? with args [u'transmission', 0]
2018-01-06 13:40:42 INFO    ::MAIN: Unable to locate download info for transmission, continuing to try and process this release ...
2018-01-06 13:40:42 DEBUG   ::MAIN: Received Directory: /media/externeHD/transmission/series/transmission | Name: transmission | Category: tv
2018-01-06 13:40:42 DEBUG   ::MAIN: SEARCH: Could not find the category: tv in the directory structure
2018-01-06 13:40:42 INFO    ::MAIN: SEARCH: Found torrent directory transmission in the directory structure
2018-01-06 13:40:42 DEBUG   ::MAIN: Determined Directory: /media/externeHD/transmission/series/transmission | Name: transmission | Category: tv
2018-01-06 13:40:42 INFO    ::MAIN: Auto-detected SECTION:NzbDrone
2018-01-06 13:40:42 INFO    ::MAIN: Output directory set to: /media/externeHD/transmission/process/tv/transmission
2018-01-06 13:40:42 DEBUG   ::MAIN: Scanning files in directory: /media/externeHD/transmission/series/transmission
2018-01-06 13:40:42 DEBUG   ::MAIN: Found 1 files in /media/externeHD/transmission/series/transmission
2018-01-06 13:40:42 INFO    ::COPYLINK: MEDIAFILE: [test.mkv]
2018-01-06 13:40:42 INFO    ::COPYLINK: SOURCE FOLDER: [/media/externeHD/transmission/series/transmission]
2018-01-06 13:40:42 INFO    ::COPYLINK: TARGET FOLDER: [/media/externeHD/transmission/process/tv/transmission]
2018-01-06 13:40:42 INFO    ::COPYLINK: Hard linking SOURCE MEDIAFILE -> TARGET FOLDER
2018-01-06 13:40:42 INFO    ::MAIN: Starting manual run for NzbDrone:tv - Folder:/media/externeHD/transmission/series/transmission
2018-01-06 13:40:42 INFO    ::MAIN: Checking database for download info for transmission ...
2018-01-06 13:40:42 DB      ::DB: Getting download info for transmission from the DB
2018-01-06 13:40:42 DB      ::MAIN: nzbtomedia.db: SELECT * FROM downloads WHERE input_name=? AND status=? with args [u'transmission', 0]
2018-01-06 13:40:42 INFO    ::MAIN: Unable to locate download info for transmission, continuing to try and process this release ...
2018-01-06 13:40:42 DEBUG   ::MAIN: Received Directory: /media/externeHD/transmission/series/transmission | Name: transmission | Category: tv
2018-01-06 13:40:42 DEBUG   ::MAIN: SEARCH: Could not find the category: tv in the directory structure
2018-01-06 13:40:42 INFO    ::MAIN: SEARCH: Found torrent directory transmission in the directory structure
2018-01-06 13:40:42 DEBUG   ::MAIN: Determined Directory: /media/externeHD/transmission/series/transmission | Name: transmission | Category: tv
2018-01-06 13:40:42 INFO    ::MAIN: Auto-detected SECTION:NzbDrone
2018-01-06 13:40:42 INFO    ::MAIN: Output directory set to: /media/externeHD/transmission/process/tv/transmission
2018-01-06 13:40:42 DEBUG   ::MAIN: Scanning files in directory: /media/externeHD/transmission/series/transmission
2018-01-06 13:40:42 DEBUG   ::MAIN: Found 1 files in /media/externeHD/transmission/series/transmission
2018-01-06 13:40:42 INFO    ::COPYLINK: MEDIAFILE: [test.mkv]
2018-01-06 13:40:42 INFO    ::COPYLINK: SOURCE FOLDER: [/media/externeHD/transmission/series/transmission]
2018-01-06 13:40:42 INFO    ::COPYLINK: TARGET FOLDER: [/media/externeHD/transmission/process/tv/transmission]
2018-01-06 13:40:42 INFO    ::COPYLINK: Hard linking SOURCE MEDIAFILE -> TARGET FOLDER
2018-01-06 13:40:42 DEBUG   ::MAIN: Checking for archives to extract in directory: /media/externeHD/transmission/series/transmission
2018-01-06 13:40:42 INFO    ::MAIN: FLATTEN: Flattening directory: /media/externeHD/transmission/process/tv/transmission
2018-01-06 13:40:42 DEBUG   ::MAIN: Checking for empty folders in:/media/externeHD/transmission/process/tv/transmission
2018-01-06 13:40:42 INFO    ::MAIN: Found 1 media files in /media/externeHD/transmission/process/tv/transmission
2018-01-06 13:40:42 INFO    ::MAIN: Calling NzbDrone:tv to post-process:transmission
2018-01-06 13:40:42 DEBUG   ::SERVER: Attempting to connect to server at http://localhost:8989
2018-01-06 13:40:42 DEBUG   ::SERVER: Server responded at http://localhost:8989
2018-01-06 13:40:42 INFO    ::MAIN: Attempting to verify tv fork
2018-01-06 13:40:43 INFO    ::MAIN: NzbDrone:tv fork set to default
2018-01-06 13:40:43 WARNING ::EXCEPTION: No name replacement determined for test
2018-01-06 13:40:43 DEBUG   ::EXCEPTION: Replacing file name /media/externeHD/transmission/process/tv/transmission/test.mkv with download name /media/externeHD/transmission/process/tv/transmission/transmission.mkv
2018-01-06 13:40:43 INFO    ::MAIN: FLATTEN: Flattening directory: /media/externeHD/transmission/process/tv/transmission
2018-01-06 13:40:43 DEBUG   ::MAIN: Checking for empty folders in:/media/externeHD/transmission/process/tv/transmission
2018-01-06 13:40:43 DEBUG   ::MAIN: calling command: /home/benjamin/bin/ffprobe -v quiet -print_format json -show_format -show_streams -show_error /home/benjamin/nzbToMedia/tests/test.mp4
2018-01-06 13:40:43 INFO    ::TRANSCODER: Checking [transmission.mkv] for corruption, please stand by ...
2018-01-06 13:40:43 DEBUG   ::MAIN: calling command: /home/benjamin/bin/ffprobe -v quiet -print_format json -show_format -show_streams -show_error /media/externeHD/transmission/process/tv/transmission/transmission.mkv
2018-01-06 13:40:43 INFO    ::TRANSCODER: SUCCESS: [transmission.mkv] has no corruption.
2018-01-06 13:40:43 INFO    ::MAIN: Checking for files to be transcoded
2018-01-06 13:40:43 DEBUG   ::MAIN: calling command: /home/benjamin/bin/ffprobe -v quiet -print_format json -show_format -show_streams -show_error /media/externeHD/transmission/process/tv/transmission/transmission.mkv
2018-01-06 13:40:43 INFO    ::MAIN: Transcoding video: /media/externeHD/transmission/series/transmission/transmission.mkv
2018-01-06 13:40:43 DEBUG   ::MAIN: calling command: nice -n17 ionice -c0 -n0 /home/benjamin/bin/ffmpeg -hwaccel vaapi -hwaccel_output_format vaapi -fflags +genpts -i /media/externeHD/transmission/process/tv/transmission/transmission.mkv -map 0:0 -map 0:1 -c:v hevc_vaapi -preset ultrafast -crf 19 -c:a:0 copy -sn /media/externeHD/transmission/series/transmission/transmission.mkv
2018-01-06 13:40:43 ERROR   ::MAIN: Transcoding of video to /media/externeHD/transmission/series/transmission/transmission.mkv failed with result 1
2018-01-06 13:40:43 ERROR   ::NZBDRONE: FAILED: Transcoding failed for files in /media/externeHD/transmission/process/tv/transmission
2018-01-06 13:40:43 ERROR   ::MAIN: A problem was reported when trying to perform a manual run for NzbDrone:tv.
2018-01-06 13:40:43 ERROR   ::MAIN: A problem was reported in the ./TorrentToMedia.py script.

i get 
::MAIN: Transcoding of video to /media/externeHD/transmission/series/transmission/transmission.mkv failed with result 1
Don't know what to do now, i don't get a specific error

i tried to set enviorment in python but that is not working

Torrenttomedia.py

> #!/usr/bin/env python2
# coding=utf-8
import datetime
import os
os.environ["DISPLAY"] = "server/unix:0"
import sys
import core

Somebody knows how i fix this?

**Comment by rosskopf** [comment]
after hours of looking i though i ask but now i found the solution
i added -report to generalOptions and ik found out that i used generalOptions wrong!

sorry about that

**Comment by clinton-hall** [comment]
so you have it working? how did you end up setting generalOptions to make this work?

**Comment by rosskopf** [comment]
I ended up with this settings


  

```
# hwAccel. 1 will set ffmpeg to enable hardware acceleration (this requires a recent ffmpeg)
    hwAccel = 0
    # generalOptions. Enter your additional ffmpeg options here with commas to separate each option/value (i.e replace spaces with commas).
    generalOptions = -hwaccel, vaapi, -hwaccel_output_format, vaapi
    # outputDefault. Loads default configs for the selected device. The remaining options below are ignored.
    # If you want to use your own profile, leave this blank and set the remaining options below.
    # outputDefault profiles allowed: iPad, iPad-1080p, iPad-720p, Apple-TV2, iPod, iPhone, PS3, xbox, Roku-1080p, Roku-720p, Roku-480p, mkv, mp4-scene-release
    outputDefault = ""
    #### Define custom settings below.
    outputVideoExtension = .mkv
    outputVideoCodec = hevc_vaapi
    VideoCodecAllow = "HEVC,hevc,h265,x265,h.265"
    outputVideoPreset = 1
    outputVideoResolution = ""
    outputVideoFramerate = ""
    outputVideoBitrate = ""
    outputVideoCRF = 21
    outputVideoLevel = 41
    outputAudioCodec = ac3
    AudioCodecAllow = "DTS,ac3,libfdk_aac,a_aac,aac,libfaac,a_ac3,ac-3"
    outputAudioChannels = ""
    outputAudioBitrate = ""
    outputAudioTrack2Codec = ""
    AudioCodec2Allow = ""
    outputAudioTrack2Channels = ""
    outputAudioTrack2Bitrate = ""
    outputAudioOtherCodec = libmp3lame
    AudioOtherCodecAllow = ""
    outputAudioOtherChannels = ""
    outputAudioOtherBitrate = 128000
    outputSubtitleCodec = ""

```
and a little edit in transcoder.py


```
       if core.VPRESET:
            video_cmd.extend(['-bf', core.VPRESET])
        if core.VCRF:
            video_cmd.extend(['-qp', str(core.VCRF)])
        if core.VLEVEL:
            video_cmd.extend(['-level', str(core.VLEVEL)])
        no_copy = ['-vf', '-crf', '-qp', '-level', '-bf', '-b:v']
        if video_cmd[1] == 'copy' and any(i in video_cmd for i in no_copy):



        map_cmd.extend(['-map', '0'])
        if core.VCODEC:
            video_cmd.extend(['-c:v', core.VCODEC])
            if core.VCODEC == 'libx264' and core.VPRESET:
                video_cmd.extend(['-pre', core.VPRESET])
        else:
            video_cmd.extend(['-c:v', 'copy'])
        if core.VFRAMERATE:
            video_cmd.extend(['-crf', str(core.VFRAMERATE)])
        if core.VBITRATE:
            video_cmd.extend(['-b:v', str(core.VBITRATE)])
        if core.VRESOLUTION:
            video_cmd.extend(['-vf', 'scale={vres}'.format(vres=core.VRESOLUTION)])
        if core.VPRESET:
            video_cmd.extend(['-bf', core.VPRESET])
        if core.VCRF:
            video_cmd.extend(['-qp', str(core.VCRF)])
        if core.VLEVEL:
            video_cmd.extend(['-level', str(core.VLEVEL)])
```

i need a little fine tune in quality of the file now, but it is working.

**Comment by pienoet** [comment]
Hi is it possible to transcode my x264 file into x265?

**Comment by clinton-hall** [comment]
@pienoet 
Yes, this should be possible.
Just ensure you don't have x264 or h264 in your `VideoCodecAllow` list. 
Then have output codec set to either:
```
outputVideoCodec = hevc_vaapi
generalOptions = -hwaccel, vaapi, -hwaccel_output_format, vaapi
``` 
as per above example, or perhaps
```
outputVideoCodec = libx265
generalOptions = -preset,ultrafast,-x265-params,lossless=1
```
The second suggestion comes from here
https://unix.stackexchange.com/questions/230800/re-encoding-video-library-in-x265-hevc-with-no-quality-loss?utm_medium=organic&utm_source=google_rich_qa&utm_campaign=google_rich_qa

**Comment by pienoet** [comment]
@clinton-hall Thank you very much this info.
Is this a general setting or can i set this only for i.e The couchpotato script?
And when does transcoding begin after moving the files in the right folder?
Is this working in arm devices too? 

Thanks!

**Comment by clinton-hall** [comment]
This is currently a general setting that applies to all categories.
It has been on my to-do list for a while to create a set of conditions on what to transcode, and this would include categories... but I have not had time to look at it (I recall starting to draft out a work flow sometime around January 2015... then I moved to a new job... then I moved Country... )

It is possible to create 2 instances of nzbToMedia (in 2 folders) and have each set up differently. Then have each category in your downloader call the different version of nzbToMedia

Transcoding happens before the movie/show is moved to the final library (By CouchPotato or SickBeard etc.) In the case of Torrents this is transcoded after doing the linking, but before moving to the final folder. So this is done in the Torrent output directory (Temporary location).

As long as your device has working Python, and working ffmpeg, this will work. I have done some testing on arm devices. The biggest issue on a small arm processor is that transcoding may take a significant length of time (but some modern arm processors are actually quite good).
